# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLLegendElement.cpp
# Commit: 6141de941e71
# Full Hash: 6141de941e717cdb5485e1aa1ca00ffcd74f8a81
# Author: Mats Palmgren <mats@mozilla.com>
# Date: 2021-01-30 19:41:15
# Regressor Bug: 1683748
# File Overlap Count: 3
# Description:
#   Bug 1683748 - Support Grid/Flex/Table/Column layout for the rendered legend of a fieldset.  r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D101555
# ==============================================================================

diff -r 3cf752580f1b -r 6141de941e71 dom/html/HTMLLegendElement.cpp
--- a/dom/html/HTMLLegendElement.cpp	Fri Jan 29 19:20:24 2021 +0000
+++ b/dom/html/HTMLLegendElement.cpp	Sat Jan 30 13:47:10 2021 +0000
@@ -103,6 +103,26 @@
   return NS_SUCCEEDED(rv.StealNSResult());
 }
 
+HTMLLegendElement::LegendAlignValue HTMLLegendElement::LogicalAlign(
+    mozilla::WritingMode aCBWM) const {
+  const nsAttrValue* attr = GetParsedAttr(nsGkAtoms::align);
+  if (!attr || attr->Type() != nsAttrValue::eEnum) {
+    return LegendAlignValue::InlineStart;
+  }
+
+  auto value = static_cast<LegendAlignValue>(attr->GetEnumValue());
+  switch (value) {
+    case LegendAlignValue::Left:
+      return aCBWM.IsBidiLTR() ? LegendAlignValue::InlineStart
+                               : LegendAlignValue::InlineEnd;
+    case LegendAlignValue::Right:
+      return aCBWM.IsBidiLTR() ? LegendAlignValue::InlineEnd
+                               : LegendAlignValue::InlineStart;
+    default:
+      return value;
+  }
+}
+
 already_AddRefed<HTMLFormElement> HTMLLegendElement::GetForm() {
   return do_AddRef(GetFormElement());
 }